DeezyMatch/rnn_networks.py

```diff
@@ -879,13 +879,13 @@ def inference(model_path, dataset_path, train_vocab_path, input_file_path,
                        dl_inputs["preprocessing"]["lowercase"],
                        dl_inputs["preprocessing"]["strip"],
                        dl_inputs["preprocessing"]["only_latin_letters"],
-                       dl_inputs["preprocessing"]["prefix_suffix"],
                        ),
         max_seq_len=dl_inputs['gru_lstm']['max_seq_len'],
         mode=dl_inputs['gru_lstm']['mode'],
         cutoff=test_cutoff, 
         save_test_class=path_save_test_class,
-        csv_sep=dl_inputs['preprocessing']["csv_sep"]
+        csv_sep=dl_inputs['preprocessing']["csv_sep"],
+        one_column_inp=True
```

Comment:
Hi @kasra-hosseini, I am looking at this PR now, thanks for taking care of it!

I believe `one_column_inp` should only be `True` at inference time when `inference_mode` is not `test` (otherwise it needs the three-column dataset format, to assess the quality of the matching, right?). I think we should raise an error/warning in this case.

See the screenshot of how it works at the moment (misleading, it seems the DeezyMatch model works really bad):
![Screenshot 2022-01-25 at 10 25 31](https://user-images.githubusercontent.com/46483603/150959358-67a46ce7-303f-4f73-b58f-8807f5c81782.png)

The `pred_results_xxxxxx.txt` file looks like this:
```
# s1	s2	prediction	p0	p1	label
Ward Hill, radar station	tmp	0	1.0000	0.0000	1
Mousa light station	tmp	0	0.8848	0.1152	1
Mablethorpe railway station	tmp	0	0.9994	0.0006	1
Theddlethorpe railway station	tmp	0	1.0000	0.0000	1
```

Reply:
😟  Such a terrible mistake! Thanks for raising this. You are completely right. I am on it.

Reply:
@mcollardanuy Is this OK? https://github.com/Living-with-machines/DeezyMatch/pull/112/commits/e23715b16535fa1717ebd7584ef8bcb5ae66e66d I also added a test for model inference with one column input, in that case, DM should raise an IndexError.

Reply:
Hi @kasra-hosseini, thanks, perfect, this makes sense!

Fyi, I am now looking into the whole issue in more detail, I am not super sure about [this](https://github.com/Living-with-machines/DeezyMatch/blob/e23715b16535fa1717ebd7584ef8bcb5ae66e66d/DeezyMatch/data_processing.py#L202):
```
# If one_column_inp is set to True, extend the row
if (len(tmp_split_row) != 3) and (one_column_inp == True):
    tmp_split_row.extend(["tmp", "true"])
```

I'll try some things and let you know!

Reply:
Right, that part can result in some unwanted behaviour. Maybe a better way is to say, if `one_column_inp` is True, then ignore all the other columns (if there is any), use only the first one, and extend that with `tmp_split_row.extend(["tmp", "true"])`

Reply:
but let me know what you suggest based on your tests.

Reply:
Yes, I think that solution would be better. I was thinking of situations in which you might care to keep more information about the queries/candidates, such as the source where you obtained them, which you could do in additional columns. Could be extend the row by inserting "tmp" and "True" as 2nd and 3rd column, instead of appending them at the end?

Reply:
Yes, exactly. I will take of this today.

Reply:
Thanks!

Reply:
@mcollardanuy I could have done this in the morning, but anyway, this is implemented here: https://github.com/Living-with-machines/DeezyMatch/pull/112/commits/180b3ae646d6a37e5ff423fbfc9705455a6b731e